src/Algebra/Morphism/RawMagma.agda

```diff
@@ -0,0 +1,111 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Consequences of a morphism between magmas
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra
+open import Algebra.Morphism
+open RawMagma using (Carrier; _≈_)
+
+module Algebra.Morphism.RawMagma
+  {a b ℓ₁ ℓ₂} {From : RawMagma b ℓ₂} {To : RawMagma a ℓ₁}
+  {to : Carrier From → Carrier To}
+  (isRawMagmaMorphism : IsRawMagmaMorphism From To to)
+  {from : Carrier To → Carrier From}
+  (from-to : ∀ x → _≈_ From (from (to x)) x)
+  (from-cong : ∀ {x y} → _≈_ To x y → _≈_ From (from x) (from y))
+  where
```

Comment:
I think `from`, `from-to` and `from-cong` can be replaced by `(to-injective : ∀ {x y} → _≈_ To (to x) (to y) → _≈_ From x y)`.

Reply:
Hmm I can't see how to prove the results without a left inverse. Maybe I'm being slow! How would the proof for `comm-homo` look?

Reply:
```
comm-homo : Commutative T._≈_ _⊕_ → Commutative F._≈_ _∙_
comm-homo comm x y = to-injective $ begin
  to (x ∙ y)  ≈⟨ ∙-homo x y ⟩
  to x ⊕ to y ≈⟨ comm (to x) (to y) ⟩
  to y ⊕ to x ≈˘⟨ ∙-homo y x ⟩
  to (y ∙ x)  ∎
  where open SetoidReasoning T-setoid
```

Reply:
Haha, oh yes. Obvious now you point it out! Thanks for the spot :smile: